File: backend/dataset/rotate/rotate_12.py
class Solution:
    def rotate(self, nums:List[int], k:int)->None:
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        nums1 = nums[:len(nums)-k]
        nums2 = nums[ len(nums)-k: ]
        
        k=k%len(nums)
        
        for i in range(len(nums2)):
            nums[i] =  nums2[i]
        for i in range(len(nums1)):
            nums[ k+i ] = nums1[i] 
        
        return nums

File: backend/CDataset/rotate/rotate_12.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int l = nums.size(), s = 0;
        while (l-s > 1 && k) {
            k %= (l-s);
            for (int i=0; i<k; ++i) {
                nums[s+i] ^= nums[l-k+i];
                nums[l-k+i] ^= nums[s+i];
                nums[s+i] ^= nums[l-k+i];
            }
            s += k;
        }
    }
};
